### ISSUE TITLE
CustomLoss produces ValueError("No variables provided.") when using only loss_inputs

### ISSUE THREAD

**Issue by ManuelZierl** [ray-project/ray#9904]
By writing a CustomLoss I get an `ValueError("No variables provided.")`.

I wanted to write my own loss model, but it seems that I cannot use the tensors given to the `custom_loss` function via the `loss_inputs` parameter. Let me show you a very simple example

```python
class TestLossModel(TFModelV2):
    def __init__(self, obs_space, action_space, num_outputs, model_config,
                 name):
        super().__init__(obs_space, action_space, num_outputs, model_config,
                         name)
        self.fcnet = FullyConnectedNetwork(
            self.obs_space,
            self.action_space,
            num_outputs,
            model_config,
            name="fcnet")
        self.register_variables(self.fcnet.variables())

    @override(ModelV2)
    def forward(self, input_dict, state, seq_lens):
        return self.fcnet(input_dict, state, seq_lens)

    @override(ModelV2)
    def custom_loss(self, policy_loss, loss_inputs):
        print("FYI: You can also use these tensors: {}, ".format(loss_inputs))
        return policy_loss + tf.reduce_mean(tf.cast(loss_inputs["actions"], tf.float32))
```

This custom loss adds an second loss term to the policy loss that is  the reduced mean of the observation (not really useful but just to demonstrate my point). Running with this loss-modell works, but if I change the return in `custom_loss` to just `return tf.reduce_mean(tf.cast(loss_inputs["actions"], tf.float32))` (so without the policy_loss) I get an `ValueError("No variables provided.")`.

This makes me very suspicious. I assume that if TensorFlow doesn't find any Variable for the `tf.reduce_mean` - term this should mean that it has actually no effect to the policy loss (even if the number changes).

Maybe I misunderstood something here too but the example custom_loss-file https://github.com/ray-project/ray/blob/master/rllib/examples/models/custom_loss_model.py (line 55) states that `print("FYI: You can also use these tensors: {}, ".format(loss_inputs))`. Isn't it that what I'm doing?

This Issue might be related #4517 but the author of this issue states that ` "hiddens": []` . Also the author overrides `_build_layers_v2` which I didn't.

Here is also my Setup + Environment, even if I do not think that it has something to do with it since i have tried it with multiple examples

<details>
  <summary>Enviroment + Setup</summary>

```python
import random
import time
from logging import warning

from gym.spaces import Discrete, Box
import numpy as np

import ray
from ray import tune
from ray.rllib.env.multi_agent_env import MultiAgentEnv
from ray.tune import register_env, grid_search
from ray.rllib.agents.dqn.dqn_tf_policy import DQNTFPolicy
from ray.rllib.models import ModelCatalog
from custom_loss import  TestLossModel
import json


class TestGame(MultiAgentEnv):
    def __init__(self, env_config):
        self.counter = 0
        self.max_len = 1000


    def reset(self):
        self.channel_2 = 0
        return {0:np.array([0])}

    def step(self, action_dict):
        rewards = {i: 0 for i in range(len(action_dict))}
        dones = {"__all__": False}
        self.counter += 1
        if self.counter >= self.max_len :
            dones = {"__all__": True}
            self.counter = 0

        obs = {0: np.array([action_dict[0]])}
        return obs, rewards, dones, {}


if __name__ == "__main__":
    run = "DQN"

    obs_space = Box(low=np.array([0]), high=np.array([11]))
    act_space = Discrete(10)

    mapping = {0: "pol_0"}
    policies = {
        "pol_0" : (None, obs_space, act_space, {"agent_id": 0}),
        }

    policy_name_mapping = lambda x: mapping[x]

    config = {
        "learning_starts": 100,
        "evaluation_interval": 500//100,
        "evaluation_num_episodes": 5,
        "evaluation_config": {
            "explore": False,
            "env_config": {
                "is_testing": True,
            }
        },
        "env_config": {
            "logits": False,
        },
        "multiagent": {
            "policies": policies,
            "policy_mapping_fn": policy_name_mapping,
        },
        "num_workers": 1
    }


    ModelCatalog.register_custom_model("custom_loss", TestLossModel)
    config["model"] = {
        "custom_model": "custom_loss",
        "custom_model_config": {
                        "channels": Discrete(3),
        },
    }

    ray.init(
        num_cpus=4,
    )
    tune.run(
        run,
        stop={
            "timesteps_total": 500000,
        },
        config=dict(config, **{
            "env": TestGame,
        }),
    )
``` 
  
</details>

software | version
--- | ---
ray |0.8.4
python | 3.6.9
tensorflow | 1.14.0
OS |  Ubuntu
